lib/projects.js

```diff
@@ -0,0 +1,31 @@
+const { collection } = require("./database");
+const { ObjectID } = require("mongodb");
+
+async function setProject({ data, tags }) {
+  await collection("projects").insertOne({ data, tags });
+}
+
+async function getProjects() {
+  return await collection("projects").find({}).sort({ _id: -1 }).toArray();
+}
+
+async function getProjectbyId(id) {
+  const objectId = new ObjectID.createFromHexString(id);
+  return await collection("projects").findOne({ _id: objectId });
+}
+async function getProjectbyTag(tag) {
+  return await collection("projects").find({ tags: tag }).toArray();
+}
+async function getCategory(category) {
+  return await collection("projects")
+    .find({
+      "data.category": category,
```

Comment:
Consider reworking the schema of `projects` and replace `data.category` with `category`.

Reply:
data comes from the react-hook-form, not sure if i can change this

Reply:
It's about these lines?
https://github.com/kschmitz123/diy-project-app/blob/mongoDB/client/src/components/UploadProject.js#L37-L49

You have several options to solve it. I would create a `project` variable here.

```js
  const onSubmit = async (data) => {
    const tags = data.tags.match(/[^,\s?]+/g);
    const newProject = {
      ...data,
      tags
    };
    try {
      setLoading(true);
      const project = await postProject(project);
      setLoading(false);
      if (!previewSource) return;
      uploadImage(previewSource);
      history.push(`/projects/${project.id}`);
    } catch (error) {
      setErrorMessage(error.message);
    }
  };
```